src/main/java/Duke.java

```diff
@@ -1,10 +1,249 @@
+                    task.markAsNotDone();
+                    System.out.println("OK, I've marked this task as not done yet:");
+                    System.out.println(task);
+                } else {
+                    System.out.println("Task is already marked as not done.");
+                }
+                found = true;
+                break;
+            }
+        }
+
+        if (!found) {
+            System.out.println("Task not found.");
+        }
+    }
+
+    public void echoCommands() {
+        Scanner scanner = new Scanner(System.in);
+        String command;
+
+        do {
+            command = scanner.nextLine();
+            if (command.equals("list") || command.startsWith("list ")) {
+                listTasks();
+            } else if (command.startsWith("mark ")) {
+                markTaskAsDone(command.substring(5));
+            } else if (command.startsWith("unmark ")) {
+                unmarkTaskAsDone(command.substring(7));
+            } else if (!command.startsWith("bye")) {
+                addTask(command);
+            }
+        } while (!command.equals("bye"));
+
+        scanner.close();
+    }
+
+    public void exit() {
+        System.out.println("Bye. Hope to see you again soon!");
+        System.out.println("=========================");
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        Duke chatbot = new Duke("aoliba");
+        chatbot.greet();
+        chatbot.echoCommands();
+        chatbot.exit();
+    }
+
+    private abstract class Task {
+        private final String description;
+        private boolean isDone;
+        public abstract String getType();
+
+        public Task(String description) {
+            this.description = description;
+            this.isDone = false;
+        }
+
+        public void markAsDone() {
+            this.isDone = true;
+        }
+
+        public void markAsNotDone() {
+            this.isDone = false;
+        }
+
+        public String getDescription() {
+            return description;
+        }
+
+        public boolean isDone() {
+            return isDone;
+        }
+
+        @Override
+        public String toString() {
+            String status = isDone ? "[X]" : "[ ]";
+//            return status + " " + description;
+            return "[" + getType() + "]" + status + " " + description;
+        }
+
+    }
+
+    public class Todo extends Task {
+
+        public Todo(String description) {
+            super(description);
+        }
+
+        @Override
+        public String getType() {
+            return "T";
+        }
+    }
+
+    public class Deadline extends Task {
+
+        private String ddl;
```

Comment:
Avoid using variable name that can be confusing. Can just set it to deadline instead.